src/main/java/net/minecraftforge/common/conditions/ICondition.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.conditions;
+
+import net.minecraft.core.Holder;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.TagKey;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * An {@link ICondition} represents any type of loading condition that can be applied to a JSON object.
+ */
+public interface ICondition extends Predicate<ICondition.IContext>
+{
+	/**
+	 * @return The ID of the serializer that is responsible for this condition.
+	 */
+    ResourceLocation getSerializerId();
+
+    /**
+     * Tests if this condition is met, given the context.
+	 * @param context The {@linkplain ICondition.IContext Condition Context}
```

Comment:
Formatting: 4 spaces, not tabs.
```suggestion
    /**
     * @return The ID of the serializer that is responsible for this condition.
     */
    ResourceLocation getSerializerId();

    /**
     * Tests if this condition is met, given the context.
     * @param context The {@linkplain ICondition.IContext Condition Context}
```